pennylane/measurements.py

```diff
@@ -445,16 +591,31 @@ def circuit(x):
             f"{op.name} is not an observable: cannot be used with sample"
         )
 
+    try:
+        eigvals = qml.eigvals(op)
+        if op is None or all(np.issubdtype(e.dtype, int) for e in eigvals):
```

Comment:
Is it possible to have `op=None` at this stage?

Reply:
This is now outdated (though should be possible, yes).